lib/ruby_indexer/lib/ruby_indexer/index.rb

```diff
@@ -237,20 +237,20 @@ def follow_aliased_namespace(name)
       real_parts.join("::")
     end
 
-    # Attempts to find a given method for a resolved fully qualified receiver name. Returns `nil` if the method does not
-    # exist on that receiver
-    sig { params(method_name: String, receiver_name: String).returns(T.nilable(Entry::Member)) }
+    # Attempts to find methods for a resolved fully qualified receiver name.
+    # Returns an empty array if the method does not exist on that receiver
+    sig { params(method_name: String, receiver_name: String).returns(T::Array[Entry::Member]) }
     def resolve_method(method_name, receiver_name)
       method_entries = self[method_name]
       owner_entries = self[receiver_name]
-      return unless owner_entries && method_entries
+      return [] unless owner_entries && method_entries
```

Comment:
I think I would keep the nilable return. The reasoning being that Sorbet will be able to tell us the right type after an early return ([example](https://sorbet.run/#%23%20typed%3A%20true%0Aextend%20T%3A%3ASig%0A%0Asig%20%7Breturns%28T.nilable%28T%3A%3AArray%5BInteger%5D%29%29%7D%0Adef%20foo%0Aend%0A%0Ar%20%3D%20foo%0AT.reveal_type%28r%29%0Areturn%20unless%20r%0A%0AT.reveal_type%28r%29)).

With arrays only, we would need to remember to check if they are empty everywhere.

Reply:
TIL. I'm not so familiar with Sorbet. 👍  

Let me know if it's OK for you.

Reply:
It looks great!